src/core/server/http/http_service.ts

```diff
@@ -117,6 +116,22 @@ export class HttpService
       ) => this.requestHandlerContext!.registerContext(pluginOpaqueId, contextName, provider),
     };
 
+    if (notReadySetup) {
+      this.log.warn('haz not ready');
+      this.internalSetup.notReadyServer = {
+        registerRoutes: (path: string, registerCallback: (router: IRouter) => void) => {
+          const router = new Router(
+            path,
+            this.log,
+            this.requestHandlerContext!.createHandler.bind(null, this.coreContext.coreId)
+          );
+
+          registerCallback(router);
```

Comment:
This callback + the `allowAfterListening` flag on `registerRouter` are the 'hackiest' parts of this solution. Basically it allows the status service to register a route with the not ready server after it's started listening. This has the benefit of starting the not ready server ASAP while still being able to serve this status route before Kibana finished migrations.